account_asset_management_no_depreciation/models/account_asset_profile.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountAssetProfile(models.Model):
+    _inherit = "account.asset.profile"
+
+    no_depreciation = fields.Boolean(
+        string="Non-Depreciation",
+        help="Check this if you do not want to compute depreciation.",
+    )
+
+    @api.onchange("no_depreciation")
+    def _onchange_no_depreciation(self):
+        if not self.no_depreciation:
+            self.method_number = 0
+
+    @api.constrains("no_depreciation", "method_number")
+    def _check_method(self):
+        if any(a.no_depreciation and a.method_number != 0 for a in self):
+            raise UserError(
+                _("Non-Depreciation is only supported for Number of Years = 0.")
+            )
```

Comment:
I think if we require method_number == 0, we don't need following.
```suggestion
```

Reply:
Deleted.